WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift

```diff
@@ -0,0 +1,252 @@
+//  WordPress
+//
+//  Created by Brandon Titus on 1/7/20.
+//  Copyright © 2020 WordPress. All rights reserved.
+//
+
+import Foundation
+import WebKit
+import Gridicons
+
+class NewPostPreviewViewController: UIViewController {
+    var webView: WKWebView!
+
+    var onClose: (() -> Void)? = nil
+
+    let post: AbstractPost
+
+    private let generator: PostPreviewGenerator
+    private var reachabilityObserver: Any?
+
+    private weak var noResultsViewController: NoResultsViewController?
+
+    init(post: AbstractPost, previewURL: URL? = nil) {
+        self.post = post
+        generator = PostPreviewGenerator(post: post, previewURL: previewURL)
+        super.init(nibName: nil, bundle: nil)
+        generator.delegate = self
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private lazy var statusButtonItem: UIBarButtonItem = {
+        let statusView = LoadingStatusView(title: NSLocalizedString("Loading", comment: "Label for button to present loading preview status"))
+        let buttonItem = UIBarButtonItem(customView: statusView)
+        buttonItem.accessibilityIdentifier = "Preview Status"
+        return buttonItem
+    }()
+
+    private lazy var doneBarButtonItem: UIBarButtonItem = {
+        let buttonItem = UIBarButtonItem(title: NSLocalizedString("Done", comment: "Label for button to dismiss post preview"), style: .done, target: self, action: #selector(dismissPreview))
+        buttonItem.accessibilityIdentifier = "Done"
+        return buttonItem
+    }()
+
+    private lazy var shareBarButtonItem: UIBarButtonItem = {
+        let image = Gridicon.iconOfType(.shareIOS)
+        let buttonItem = UIBarButtonItem(image: image, style: .plain, target: self, action: #selector(sharePost))
+        buttonItem.accessibilityLabel = NSLocalizedString("Share", comment: "Title of the share button in the Post Editor.")
+        buttonItem.accessibilityIdentifier = "Share"
+        return buttonItem
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupWebView()
+        setupBarButtons()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        refreshWebView()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        stopWaitingForConnectionRestored()
+    }
+
+    private func refreshWebView() {
+        generator.generate()
+    }
+
+    private func setupBarButtons() {
+        navigationItem.rightBarButtonItems = [doneBarButtonItem, shareBarButtonItem]
+        navigationItem.leftItemsSupplementBackButton = true
+    }
+
+    private func setupWebView() {
+        webView = WKWebView(frame: view.bounds)
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+        view.pinSubviewToAllEdges(webView)
+    }
+
+    @objc private func dismissPreview() {
+        onClose?()
```

Comment:
This is a slightly different behaviour to the existing view controller (not sure if that matters?). The previous one only called dismiss on the presenting view controller if there was no `onClose` block provided.